packages/backend/src/registries/LocalRepositoryRegistry.ts

```diff
@@ -36,16 +42,35 @@ export class LocalRepositoryRegistry extends Publisher<LocalRepository[]> {
     this.notify();
 
     return Disposable.create(() => {
-      this.unregister(localRepository.path);
+      this.unregister(localRepository.path).catch((reason: unknown) => console.log(String(reason)));
     });
   }
 
-  unregister(path: string): void {
+  async unregister(path: string): Promise<void> {
+    await this.deleteLocalRepository(path);
```

Comment:
I'm a little afraid of having this added to the `unregister` method. Up to now, register/unregister seems to be related to stuff in memory, but has no side effect on disk.

Also, the `unregister` method is called as part of the "disposal" of the `register` method, but this is not called for now in the code. There is a risk that it is added later, thinking it is just about clearing things in memory or temporary things (timers for example).